docs/test-architecture.md

```diff
@@ -0,0 +1,61 @@
+Testing via Helm
+================
+
+Problem Summary
+---------------
+
+Currently in helm/charts we have a simple way of health-checking charts which
+checks that all pods in the chart reach "Running" state on a running Kubernetes
+cluster.In order to build a system that holds chart quality at the forefront,
+we need to introduce an easy way to create and run chart-centric tests.
```

Comment:
Rather than merely observe that all pods in the chart reach "Running" state, it would be better to require that they have a livenessProbe, and that it return 200 OK.